### PR TITLE
CH23169 add desired ad in header

### PR DESCRIPTION
**Pull request by calliebensel**

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -159,7 +160,8 @@ class DecisionClient {
                 'in production environments.\n' +
                 '--------------------------------------------------------------'
             );
-            headers['x-adzerk-explain'] = additionalOpts.apiKey || '';
+            headers['x-adzerk-explain'] =
+              additionalOpts.apiKey || JSON.stringify(additionalOpts.desiredAd) || '';
```
Comment (CrshOverride) [inline src/client.ts:164]: This still isn't going to work. There's no api key included if there's a desired ad. And there's still no support for desired ad map. Please look at numbers 2, 3, and 4 in https://app.shortcut.com/adzerk/story/23169/update-sdks-to-support-explainer-2-0-input-output for all of the supported JSON formats for the `x-adzerk-explain` header.

Comment (calliebensel) [reply src/client.ts]: Updated

File: src/client.ts [hunk 2]
```diff
@@ -67,6 +67,7 @@ interface AdditionalOptions {
   userAgent?: string;
   includeExplanation?: boolean;
   apiKey?: string;
+  desiredAd?: [];
```
[the remark this answers is not in the document]
Comment (calliebensel) [reply src/client.ts]: Updated

File: package-lock.json [hunk 3]
```diff
@@ -7757,8 +7757,6 @@
     },
     "node-forge": {
       "version": "0.10.0",
-      "resolved": "https://registry.npmjs.org/node-forge/-/node-forge-0.10.0.tgz",
```
Comment (CrshOverride) [inline package-lock.json:7760]: This needs to be reverted as it will cause Dependabot to complain in the future.

Comment (calliebensel) [reply package-lock.json:7760]: fixed!

**Review by CrshOverride — APPROVED** [review]

**Pull request merged**